nacl-fast.js

```diff
@@ -2361,17 +2361,16 @@ nacl.setPRNG = function(fn) {
   // Initialize PRNG if environment provides CSPRNG.
   // If not, methods calling randombytes will throw.
   var crypto;
-  if (typeof window !== 'undefined') {
-    // Browser.
-    if (window.crypto && window.crypto.getRandomValues) {
-      crypto = window.crypto; // Standard
-    } else if (window.msCrypto && window.msCrypto.getRandomValues) {
-      crypto = window.msCrypto; // Internet Explorer 11+
-    }
-    if (crypto) {
+  if (typeof process === 'undefined' || process.browser) {
```

Comment:
There is long living nw.js issue about this:
https://github.com/nwjs/nw.js/issues/494
So I think that's not available. Can't check now.

2016-02-20 18:54 GMT+10:00 Dmitry Chestnykh <notifications@github.com>:

> In nacl-fast.js
> <https://github.com/dchest/tweetnacl-js/pull/87#discussion_r53547482>:
>
> > @@ -2361,17 +2361,16 @@ nacl.setPRNG = function(fn) {
> >    // Initialize PRNG if environment provides CSPRNG.
> >    // If not, methods calling randombytes will throw.
> >    var crypto;
> > -  if (typeof window !== 'undefined') {
> > -    // Browser.
> > -    if (window.crypto && window.crypto.getRandomValues) {
> > -      crypto = window.crypto; // Standard
> > -    } else if (window.msCrypto && window.msCrypto.getRandomValues) {
> > -      crypto = window.msCrypto; // Internet Explorer 11+
> > -    }
> > -    if (crypto) {
> > +  if (typeof process === 'undefined' || process.browser) {
>
> The new logic will be:
>
>  if (typeof process === 'undefined' || process.browser ||
>       (typeof window !== 'undefined' && window.self === self)) {
>
> Do you know if WebWorkers in Electron/nw have process defined?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dchest/tweetnacl-js/pull/87/files#r53547482>.
>